presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/EliminateCrossJoins.java

```diff
@@ -50,7 +50,7 @@
 public class EliminateCrossJoins
         implements Rule
 {
-    private static final Pattern PATTERN = Pattern.node(JoinNode.class);
+    private static final Pattern PATTERN = Pattern.matchByClass(JoinNode.class);
```

Comment:
Please add a `Patterns` or `PlanNodes` domain-level DSL class having static no-arg methods for all the node types used in patterns so far, so that this line looks like:

`private static final Pattern PATTERN = join();`

It's much less clutter and moves us closer to the pattern matching design doc. Plus, after that change, further usages created by other devs will already be done in their target form.